bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/MapValueImpl.java

```diff
@@ -271,6 +271,9 @@ public V put(K key, V value) {
 
     protected void populateInitialValues(BMapInitialValueEntry[] initialValues) {
         for (BMapInitialValueEntry initialValue : initialValues) {
+            if (initialValue == null) {
```

Comment:
IMO we shouldn't do this. We should create and pass the initialValues array only if the values are present.

Reply:
I have made a such change because there are some  cases(reale to [this](https://github.com/ballerina-platform/ballerina-lang/blob/d042e7043f64f09138932557b2b7be435000542f/bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/ErrorUtils.java#L54)) which only the detail become null. Is there any other option to handle this ?

Reply:
I think we should refactor that code to create an array of the exact length depending on how many entries there are, instead of always creating an array of length 2.

Reply:
Fixed.